meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -211,51 +178,52 @@ function fillTabletData(clinicObj) {
     });
 }
 
-//This method will handle all the data related to week columns "tot and reg"
-function buildWeekData(clinicObj, minuWeek, colKey) {
-    var weekVal = week - minuWeek;
-    $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
-        $.each(tabletDataArray, function(index, value) {
-            if (value.DevicesIDs === "- DeviceId  " + deviceVal.deviceId) {
-                var itemName = colKey + weekVal;
-                value[itemName] = deviceVal.submissionsCount;
-            }
-        });
-    });
-}
 
+//this method will handle to fill the data from the apis into the table
+function fillAPIResultData(clinicObj, minusNum, colKey) {
 
-//This method will handle all the data related to year columns
-function buildYearData(clinicObj, minuYear, colKey) {
-    var yearVal = current_Year - minuYear;
-    $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
-        $.each(tabletDataArray, function(index, value) {
-            if (value.DevicesIDs === "- DeviceId  " + deviceVal.deviceId) {
-                var itemName = colKey + yearVal;
-                value[itemName] = deviceVal.submissionsCount;
-            }
+    //This section will handle all the data related to week columns "tot and reg"
+    if (colKey == "casesInWeek" || colKey == "noOfRegistersInWeek") {
+        var weekVal = week - minusNum;
+        $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
+            $.each(tabletDataArray, function(index, value) {
+                if (value.DevicesIDs === "DeviceId " + deviceVal.deviceId) {
+                    var itemName = colKey + weekVal;
+                    value[itemName] = deviceVal.submissionsCount;
+                }
+            });
         });
-    });
+    } else if (colKey == "casesInYear") {
+        //This section will handle all the data related to year columns
+        var yearVal = current_Year - minusNum;
+        $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
+            $.each(tabletDataArray, function(index, value) {
+                if (value.DevicesIDs === "DeviceId " + deviceVal.deviceId) {
+                    var itemName = colKey + yearVal;
+                    value[itemName] = deviceVal.submissionsCount;
+                }
+            });
+        });
+    } else if (colKey == "casesInMonthL" || colKey == "casesInMonthC") {
+        //This method will handle all the data related to month columns
+        var monthVal = (new Date()).getMonth() + 1 - minusNum; // +1 because this method return index for month jan = 0 & dec =11
```

Comment:
This comment is helpful. However in my opinion unnecessarily too long.
`// + 1 because index of Jan = 0` sounds same with less words.
It's hard to tell what do you refer to by "this method".

Reply:
edited